Paintroid/src/main/java/org/catrobat/paintroid/dialog/SaveBeforeLoadImageDialog.java

```diff
@@ -60,7 +60,7 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
 					@Override
 					public void onClick(DialogInterface dialog, int id) {
 						MainActivityContracts.Presenter presenter = activity.getPresenter();
-						presenter.saveImageConfirmClicked(requestCode, uri);
+						presenter.checkPermissionAndForward(requestCode, uri);
```

Comment:
this seems to translate SaveImageRequestCode from to PermissionRequestCode - please do this translation in the presenter.